src/components/modals/CreateRepoModalTemplate.tsx

```diff
@@ -0,0 +1,265 @@
+import {
+  Modal,
+  ModalVariant,
+  Button,
+  Form,
+  FormGroup,
+  TextInput,
+  Radio,
+  SelectVariant,
+  Select,
+  SelectOption,
+  Grid,
+  Alert,
+  Divider,
+} from '@patternfly/react-core';
+import {useRecoilValue} from 'recoil';
+import {UserOrgs, UserState} from 'src/atoms/UserState';
+import {
+  createNewRepository,
+  RepositoryCreationResponse,
+} from 'src/resources/RepositoryResource';
+import {useRef, useState} from 'react';
+import {AxiosResponse} from 'axios';
+import {RepositoryListProps} from 'src/routes/RepositoriesList/RepositoriesList';
+
+enum visibilityType {
+  PUBLIC = 'PUBLIC',
+  PRIVATE = 'PRIVATE',
+}
+
+export const CreateRepositoryModalTemplate = (
+  props: CreateRepositoryModalTemplateProps,
+): JSX.Element => {
+  const {isModalOpen, handleModalToggle, orgNameProp, updateListHandler} =
+    props;
+
+  const userOrgs = useRecoilValue(UserOrgs);
+  const loggedInUser = useRecoilValue(UserState);
+
+  const [newRepository, setNewRepository] = useState({
+    name: '',
+    description: '',
+  });
+
+  const [currentOrganization, setCurrentOrganization] = useState({
+    name: orgNameProp !== null ? orgNameProp : null,
+    isDropdownOpen: false,
+  });
+
+  const [repoVisibility, setrepoVisibility] = useState(visibilityType.PUBLIC);
+
+  const nameInputRef = useRef();
+
+  const handleNameInputChange = (value) => {
+    setNewRepository({...newRepository, name: value});
+  };
+
+  const handleRepoDescriptionChange = (value) => {
+    setNewRepository({...newRepository, description: value});
+  };
+
+  // const orgsDropdown = () => {
+  //   const children = userOrgs.map((orgs, idx) => (
+  //     <SelectOption key={idx} value={orgs.name} />
+  //   ));
+  //   return [
+  //     ...children,
+  //     <Divider component="li" />,
+  //     <SelectOption>{loggedInUser} </SelectOption>,
+  //   ];
+  // };
+
+  const createRepositoryHandler = async () => {
+    handleModalToggle(); // check if this is needed
```

Comment:
Do you want to remove the comment if you've checked this?

Reply:
turns out, it's needed